### PR TITLE
Improve UX when completing struct

### PR DESCRIPTION
**Pull request by scottming**
Fixes the first part of #182 

This PR has two improvements:

The first one is related to 182 issues. Additionally, I don't think we need fuzzy search. For example, if the target is `%Document.Lines{}`, without fuzzy search, we can simply type `%Doc` and then press enter or tab for auto-completion. More options will appear automatically and we can filter by typing `l`. With fuzzy search, maybe we could complete it by typing %Docl, but you need to remember that this module exists in advance. I don't think this experience is good. The current way is already very convenient for me.

~~The second improvement adds structure documentation, including **fields**, **types** and **default values**. I really like this UI but I'm unsure if you guys like the code implementation too. I'd like to hear your opinions first before adding more tests.~~

https://github.com/lexical-lsp/lexical/assets/12830256/c3fb4355-a51f-4fdd-bee0-f5e55ed7b392

### CODE REVIEW

File: apps/remote_control/lib/lexical/remote_control/api.ex [hunk 1]
```diff
@@ -47,6 +48,10 @@ defmodule Lexical.RemoteControl.Api do
     ])
   end
 
+  def struct_doc(%Project{} = project, struct_module) do
```
Comment (scottming) [inline apps/remote_control/lib/lexical/remote_control/api.ex:51]: I think the API here could be designed to be more versatile, not just limited to struct doc.

File: apps/remote_control/lib/lexical/remote_control/completion/struct.ex [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+      _ ->
+        "#{inspect(struct.__struct__(), pretty: true, width: 40)}"
+        |> trim_parent(full_name)
+    end
+  end
+
+  defp fetch_struct_type(struct) do
+    with {:ok, types} <- Code.Typespec.fetch_types(struct),
+         {:type, t} <- Enum.find(types, &match?({:type, {:t, _, _}}, &1)) do
```
Comment (scohen) [inline apps/remote_control/lib/lexical/remote_control/completion/struct.ex:25]: this relies on the typespec being called `t`. You should look for a type that is being defined in terms of a map with a`__struct__` key.

File: apps/server/lib/lexical/server/code_intelligence/completion/translations/module_or_behaviour.ex [hunk 3]
```diff
@@ -24,15 +24,28 @@ defmodule Lexical.Server.CodeIntelligence.Completion.Translations.ModuleOrBehavi
 
     defines_struct? = Intelligence.defines_struct?(env.project, module.full_name)
 
+    immediate_descentent_structs =
+      immediate_descendent_struct_modules(env.project, module.full_name)
+
+    defines_struct_in_descentents? =
```
Comment (scohen) [inline apps/server/lib/lexical/server/code_intelligence/completion/translations/module_or_behaviour.ex:30]: `immediate_struct_in_descendents` and `immediate_descendent_structs`

File: apps/remote_control/lib/lexical/remote_control/api.ex [hunk 4]
```diff
@@ -47,6 +48,10 @@ defmodule Lexical.RemoteControl.Api do
...
+    RemoteControl.call(project, Struct, :doc, [struct_module])
```
Comment (scohen) [inline apps/remote_control/lib/lexical/remote_control/api.ex:52]: this should go through `Completion`, and it should be called `documentation` not `doc`

File: apps/remote_control/lib/lexical/remote_control/completion/struct.ex [hunk 5]
```diff
@@ -0,0 +1,80 @@
+defmodule Lexical.RemoteControl.Completion.Struct do
```
Comment (scohen) [inline apps/remote_control/lib/lexical/remote_control/completion/struct.ex:1]: The documentation is complicating the rest of this PR. We don't presently add documentation to other things, and I think we should table this

**Comment by scohen** [comment]
>  I don't think we need fuzzy search

I gave you an example of where fuzzy search comes in handy. Providing a different example where you don't think it's handy doesn't make your case. I think it's important, and we already have it for other types of completions.

I've looked at this, and it's pretty complicated for what you get. Let's not focus on documentation right now, as that's the lion's share of the complexity here.

File: apps/remote_control/lib/lexical/remote_control/completion/struct.ex (hunk 5 above)
Comment (scottming) [reply apps/remote_control/lib/lexical/remote_control/completion/struct.ex]: Ok, we can do the documentation part in another pr

File: apps/server/lib/lexical/server/code_intelligence/completion/translations/struct.ex [hunk 6]
```diff
@@ -20,6 +19,19 @@ defmodule Lexical.Server.CodeIntelligence.Completion.Translations.Struct do
...
+      kind: :module,
+      label: "#{module_name}...(#{more} more structs)",
+      detail: "#{full_name}."
+    ]
+
+    insert_text = "#{module_name}."
+    range = edit_range(env)
+
+    builder.text_edit_snippet(env, insert_text, range, builder_opts)
+  end
```
Comment (scohen) [inline apps/server/lib/lexical/server/code_intelligence/completion/translations/struct.ex:33]: I really don't like this pattern, translations are about translating an elixir sense completion into LSP. What we want here is to collapse and aggregate module completions that are returned from remote_control. Do we need another place to do this work?

**Pull request merged**